osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -33,9 +33,6 @@ internal class TimelineBlueprintContainer : EditorBlueprintContainer
         private Bindable<HitObject> placement;
         private SelectionBlueprint<HitObject> placementBlueprint;
 
-        // We want children within the timeline to be interactable
-        public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => timeline.ScreenSpaceDrawQuad.Contains(screenSpacePos);
```

Comment:
I don't follow, having `SelectionHandler` alone receiving input beyond bounds still requires `BlueprintContainer` to do the same as well, otherwise it's pointless. 

And regardless of whether `BlueprintContainer` receives input beyond bounds or not, it'll still be limited to its parent's bounds, unless it's parent also override `ReceivePositionalInputAt`, so there's not a single issue caused by making `TimelineBlueprintContainer` handle input outside bounds, as far as I'm understanding this.

I've also made it clear in the `BlueprintContainer` and `SelectionHandler` xmldoc that the point of receiving input outside bounds is only ever for interacting with blueprints that are partially (or fully, in case of timeline) outside bounds.

Reply:
If `ReceivePositionalInputAt` works hierarchically as you describe (I don't recall, haven't gone and checked), then the issue I'm having is as follows: knowing that selection handlers are children of blueprint containers, it makes perfect sense to me that the selection handler should receive all inputs that its parenting blueprint container does. But should all composers receive all the positional inputs their parents do? I'm leaning no on that one.

I don't like the change, is all, especially given that - as you said - it essentially changes nothing at best. That said I'm abstaining from judgement on this one - I'm not going to merge it, but I'm also not going to block it.

Reply:
I'm not sure how I feel about this, but I'm happy to go with the current solution and revisit the next time we run into issues as a result of how it's implemented.